src/inc/clr/fs/path.h

```diff
@@ -105,7 +129,8 @@ namespace clr
                 size_t cchBuf = *pcchBuffer;
 
                 IfFailRet(StringCchCopyExW(wzBuf, cchBuf, wzPathLeft, &wzBuf, &cchBuf, STRSAFE_NULL_ON_FAILURE));
-                IfFailRet(StringCchCatExW(wzBuf, cchBuf, wzBuf[-1] == W('\\') ? W("") : W("\\"), &wzBuf, &cchBuf, STRSAFE_NULL_ON_FAILURE));
+                const WCHAR directorySeparatorWString[] = {DirectorySeparatorChar, W('\0')};
+                IfFailRet(StringCchCatExW(wzBuf, cchBuf, wzBuf[-1] == DirectorySeparatorChar ? W("") : directorySeparatorWString, &wzBuf, &cchBuf, STRSAFE_NULL_ON_FAILURE));
```

Comment:
StringCchCopyEx modifies wzBuf to point to the terminating null character in the destination buffer after copying the source. wzPathLeft is checked for empty above, so wzBuf will be greater than the passed-in wzBuffer. wzBuf[-1] would then get the last non-null character that was copied into the destination buffer.